### PR TITLE
feat: add nomad custom docs landing

### PR DESCRIPTION
**Pull request by zchsh**
## "Ready for Review" checklist

- [x] The Vercel preview link has been added to this PR's description
- [x] The Asana task has been added to this PR's description
- [x] This PR's link has been added to the "📐 Pull Request" field on the Asana task
- [x] This Vercel preview link has been added to the "🔍 Preview" field on the Asana task
- [x] The description template has been filled in below

---

## Relevant links

- [Preview link][preview] 🔎
- [Asana task](https://app.asana.com/0/1201987349274776/1202129511371379/f) 🎟️

## What

Builds a new custom `/docs` landing page for `nomadproject.io`.

## Why

To improve the user experience of the `nomadproject.io/docs` landing page.

## How

Takes an identical approach to Vault in #178 & Consul in #264:

- Have split out `docs/[...page].tsx`, which is very similar to the former `[[...page]].tsx`
- New file `docs/index.tsx`, where the custom landing page is built

## Testing

To visit the preview product docs landing page, such as Consul's:

1. Open [the preview link][preview]
2. Select the desired product in the lower-left products dropdown. This switched the site's configuration to serve the product's dot-io site, rather than dev-portal.
3. Navigate to `/docs`

- [ ] Visit the Nomad docs landing page, and ensure the page looks and functions as expected
    - The new cards content should appear on the page
    - The sidebar should function as expected
    - Accessing versioned docs should function as expected. Note that the cards content is shown on the latest version only.

## Anything else?

Note that `nomadproject.io` has not yet been migrated to `dev-portal`.

[preview]: https://dev-portal-git-zsnomad-custom-docs-landing-hashicorp.vercel.app/docs

### CODE REVIEW

**Comment by zchsh** [comment]
Caveat here: we're not yet serving nomadproject.io from the dev-portal repo, so we'll need to make that switch first 🔀

**Comment by karl-cardenas-coding** [comment]
All changes look good to me ✅

File: src/pages/_proxied-dot-io/nomad/docs/content.json [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+        {
+          "title": "Enable Autopilot",
+          "url": "https://learn.hashicorp.com/tutorials/nomad/autopilot"
+        }
+      ]
+    }
+  ],
+  "developerCards": [
+    {
+      "title": "Plugins",
```
Comment (mikenomitch) [inline src/pages/_proxied-dot-io/nomad/docs/content.json:106]: "Plugins" feels off here - Maybe just "Nomad Pack Registry"?

File: src/pages/_proxied-dot-io/nomad/docs/content.json [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+        },
+        {
+          "title": "Write a custom Nomad Pack",
+          "url": "https://learn.hashicorp.com/tutorials/nomad/nomad-pack-writing-packs?in=nomad/nomad-pack"
+        },
+        {
+          "title": "Nomad Pack repository",
+          "url": "https://github.com/hashicorp/nomad-pack-community-registry"
+        }
+      ]
```
Comment (mikenomitch) [inline src/pages/_proxied-dot-io/nomad/docs/content.json:57]: Maybe also?

Community Registry
https://github.com/hashicorp/nomad-pack-community-registry

(or "Community Pack Registry" if it fits?)

File: src/pages/_proxied-dot-io/nomad/docs/content.json (hunk 1 above)
Comment (karl-cardenas-coding) [reply src/pages/_proxied-dot-io/nomad/docs/content.json]: Updated

File: src/pages/_proxied-dot-io/nomad/docs/content.json (hunk 2 above)
Comment (karl-cardenas-coding) [reply src/pages/_proxied-dot-io/nomad/docs/content.json:57]: Updated

**Comment by zchsh** [comment]
@BRKalow @karl-cardenas-coding Now that we've got nomadproject.io serving from the `dev-portal` repo, this is ready for final review ☺️ I think we should be good to merge this in today, unless there are any blockers 👍

**Comment by karl-cardenas-coding** [comment]
@BRKalow You may merge whenever you are ready :) 
Thanks again for all your help.

**Review by BRKalow — APPROVED** [review]
👏

**Pull request merged**